src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectPathCommandResolver.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class ProjectPathCommandResolver : ICommandResolver
+    {
+        public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
+        {
+            if (commandResolverArguments.CommandName == null 
+                || commandResolverArguments.ProjectDirectory == null)
+            {
+                return null;
+            }
+
+            var commandPath = commandResolverArguments.Environment.GetCommandPathFromRootPath(
```

Comment:
It was just a ICommandPathProvider interface that these guys could take a dependency on different implementations of.

The Template Method is fine by me.